### ISSUE TITLE
Questions about gradients in reasoning

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#5416]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question
@glenn-jocher 
How can I use backpropagation in `detect.py` to get the gradient of the input image?

**Comment by glenn-jocher** [comment]
@Zengyf-CVer I think there may be some confusion here. First of all, PyTorch computes gradients of model **weights** (not input images) with respect to a loss value. Second, with no labels there can be no loss. i.e. `loss = loss_fcn(predicted, true)` and since detect.py has no labels, gradient computation is impossible.

**Comment by Zengyf-CVer** [comment]
@glenn-jocher 
Is there such a situation:
input images without labels in `detect.py`, use the predicted maximum confidence as the loss, for example, the loss function is `loss_confMax = torch.max(conf)`, and then backpropagate loss_confMax, such as loss_confMax.backward(). This is how I did it, but there was a problem:
```shell
one of the variables needed for gradient computation has been modified by an inplace operation
```
The reason for this problem is that during backpropagation, the activation function in the model has the phenomenon of `inplace=True`, which leads to the change of variables.
I am not sure, in the val or detect stage of YOLOv5, can the loss (custom loss) be calculated in this way to find the gradient of the image? If this does not work, at what stage should the gradient of the image be obtained?

**Comment by glenn-jocher** [comment]
@Zengyf-CVer you can modify module attributes like inplace easily:

```python
for m in model.modules()
    m.attribute = value
```

I don't quite understand the above though, as like I mentioned before model gradients are only computed during training, and image gradients don't exist in this repo.
